helix-term/src/handlers/completion.rs

```diff
@@ -291,6 +299,223 @@ fn request_completion(
     });
 }
 
+fn path_completion(
+    cursor: usize,
+    text: helix_core::Rope,
+    doc: &helix_view::Document,
+) -> Option<BoxFuture<'static, anyhow::Result<Vec<CompletionItem>>>> {
+    if !doc.supports_path_completion() {
+        return None;
+    }
+
+    use helix_lsp::util::range_to_lsp_range;
+    // TODO find a good regex for most use cases (especially Windows, which is not yet covered...)
+    // currently only one path match per line is possible in unix
+    static PATH_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"((?:~)?(?:\.{0,2}/)+.*)$").unwrap());
+
+    let cur_line = text.char_to_line(cursor);
+    let begin_line = text.line_to_char(cur_line);
```

Comment:
I wonder if we should cap this to some reasonable maximum (maybe some N hundred characters) to prevent degrading the search if you have very long single lines.

Reply:
Good point, I made the limit now `-1000` compared to the current cursor.